code/game/objects/items/weapons/cigs.dm

```diff
@@ -358,6 +359,69 @@ LIGHTERS ARE IN LIGHTERS.DM
 	else
 		to_chat(user, "<span class='notice'>[src] straight out REFUSES to be lit by such uncivilized means.</span>")
 
+/obj/item/clothing/mask/holo_cigar
+	name = "holo-cigar"
+	desc = "A sleek electronic cigar imported straight from Sol. You feel badass merely glimpsing it..."
+	icon_state = "holocigaroff"
+	item_state = "cigaroff"
+	var/enabled = FALSE
+	/// Tracks how many cycles the user has been smoking for.
+	var/cycles_smoking = 0
+
+/obj/item/clothing/mask/holo_cigar/Destroy()
+	. = ..()
+	STOP_PROCESSING(SSobj, src)
+
+/obj/item/clothing/mask/holo_cigar/examine(mob/user)
+	. = ..()
+	if(enabled)
+		. += "[src] hums softly as it synthesizes nicotine."
+	else
+		. += "[src] seems to be inactive."
+
+/obj/item/clothing/mask/holo_cigar/process()
+	. = ..()
+	if(!iscarbon(loc))
+		return
+
+	var/mob/living/carbon/C = loc
+	if(C.wear_mask != src)
+		return
+
+	if(cycles_smoking == 0) //this could also be !cycles_smoking, but since 0 is a number this makes more sense imo
+		C.reagents.add_reagent("nicotine", 2)
+		cycles_smoking++
+	else
+		C.reagents.add_reagent("nicotine", REAGENTS_METABOLISM)
+		cycles_smoking++
```

Comment:
Do we need to increment this as we're only checking if it isn't 0 anyways? Could also convert this to a `TRUE/FALSE` instead

Reply:
This is how cigarettes do it and having it done this way is a bit more intuitive IMO